CHANGELOG.md

```diff
@@ -79,15 +79,85 @@ Task.Supervisor.async(
 
 ## Improved errors on binaries and evaluation
 
-TODO.
+OTP 25 improved errors on binary construction and evaluation. These improvements
+apply to Elixir as well. Before v1.14, errors when constructing binaries would
+often be hard-to-debug generic "argument errors". With OTP 25 and Elixir v1.14,
+more detail is provided for easier debugging. This work is part of [EEP
+54](https://www.erlang.org/eeps/eep-0054).
+
+```elixir
+# Before:
+int = 1
+bin = "foo"
+int <> bin
+# ** (ArgumentError) argument error
+
+# Now:
+int = 1
+bin = "foo"
+int <> bin
+# ** (ArgumentError) construction of binary failed:
+# segment 1 of type 'binary':
+# expected a binary but got: 1
+```
 
 ## Slicing with steps
 
-TODO.
+Elixir v1.12 introduced **stepped ranges**, which are ranges where you can
+specify the "step":
+
+```elixir
+Enum.to_list(1..10//3)
+#=> [1, 4, 7, 10]
+```
+
+Stepped ranges are particularly useful for numerical operations involving
+vectors and matrices (see [Nx](https://github.com/elixir-nx/nx), for example).
+However, the Elixir standard library was not making use of stepped ranges in its
+APIs. Elixir v1.14 starts to take advantage of steps with support for stepped
+ranges in `Enum.slice/2`:
+
+```elixir
+letters = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]
+Enum.slice(integers, 0..5//2)
+#=> ["a", "c", "e"]
+```
 
 ## Expression-based inspection
 
-TODO.
+In Elixir, it's conventional to implement the `Inspect` protocol for structs so
+that they're inspected with a syntax resembling this:
+
+```elixir
+IO.inspect(Version.parse("1.0.0"))
+#=> {:ok, #Version<1.0.0>}
```

Comment:
I think Version is confusing for this and not very common. Let's use MapSet instead? :)

Reply:
And with MapSet.new you can use `set = MapSet.new([1, 2, 3])` and then `MapSet.put(set, 4)` and show how the representation changes.

Reply:
Ah, that's a great callout! Will do.